app.go

```diff
@@ -28,6 +28,10 @@ type TSApp struct {
 	// Ephemeral specifies whether Tailscale nodes should be registered as ephemeral.
 	Ephemeral bool `json:"ephemeral,omitempty" caddy:"namespace=tailscale.ephemeral"`
 
+	// StateDir specifies the default state directory for Tailscale nodes.
+	// This value MUST contain a "{node}" placeholder which will be replaced with the node name.
```

Comment:
> MUST contain a "{node}"

This feels a little awkwad. Would it be enough to just allow to folks to specify a dir like `/foo/bar` and we automatically create `/foo/bar/<node-name>` for each node? I cannot think of a use case where someone would want `/foo/<node-name>/bar`, can you?

Reply:
I don't know that I had a specific use-case in mind, but was just thinking that if we're going to expose the config option, and caddy already has variable substitution support, it wouldn't be hard at all to support `{node}`.  Maybe rather than require it, we `filepath.Join` it to the end if it's not present?  Or maybe that's still too confusing to explain, and for not practical benefit.

Yeah, I'll just append the node name like you're suggesting, and we can always add `{node}` support in the future if someone really has a need for it.